users/users-router.js

```diff
@@ -2,9 +2,13 @@ const router = require("express").Router();
 
 const getAllUsers = require("./getAllUsers");
 const getUserById = require("./getUserById");
+const getAllJobListings = require("../jobListings/getAllJobListings");
+const getJobListingById = require("../jobListings/getJobListingById");
+const validateUserId = require("../middlewares/validateUserId");
 
 router.get("/", getAllUsers);
-
 router.get("/:id", getUserById);
+router.get("/:id/joblistings", validateUserId, getAllJobListings);
+router.get("/:id/joblistings/:id", getJobListingById, validateUserId);
```

Comment:
If you're getting job listings for a user it needs to be in it's own joblistings router file since it doesn't require admin permissions. The routers in admins router that relate to joblistings are there because the admin user needs admin privileges to it.